### PR TITLE
Updated minor typos in index.md

### PR DESCRIPTION
**Pull request by svilupp**
I've fixed a few minor typos, see the tracked file.

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -337,8 +337,8 @@ example the result of `@transform(df, @byrow :y = f(:x))` and
 
 ## Propagating missing values with `@passmissing`
 
-Many Julia functions to not automatically propagate missing values. For instance, 
-`parse(Int, missing)` will error. 
+Many Julia functions do not automatically propagate missing values. For instance, 
+`parse(Int, missing)` will throw an error.
```
Comment (pdeffebach) [inline docs/src/index.md:341]: I think "will error" is fine. 
```suggestion
`parse(Int, missing)` will error. 
```

File: docs/src/index.md [hunk 2]
```diff
@@ -610,13 +610,13 @@ Consequently,
 @transform(df, :y = :A + $"B")
 ```
 
-will not error even though 
+will not throw an error even though
```
Comment (pdeffebach) [inline docs/src/index.md:613]:
```suggestion
will not error even though 
```

File: docs/src/index.md [hunk 3]
```diff
@@ -610,13 +610,13 @@ Consequently,
...
 
-will not error even though 
+will not throw an error even though 
 
 ```
 transform(df, [:A, "B"] => (+) => :y)
 ```
 
-will error in DataFrames. 
+will throw an error in DataFrames.
```
Comment (pdeffebach) [inline docs/src/index.md:619]:
```suggestion
will error in DataFrames. 
```

File: docs/src/index.md (hunk 1 above)
Comment (svilupp) [reply docs/src/index.md:341]: I can leave it as is and close the PR, but I found the phrase distracting.

English is my second language, but I believe "error" is a noun ([MW on "error"](https://www.merriam-webster.com/dictionary/error#synonyms)) and the verb would be "to err" ([MW on "err"](https://www.merriam-webster.com/dictionary/err). 

I've also searched Google for the phrases "to error" and "will error" and haven't found any reputable reference.

**Comment by bkamins** [comment]
I am not a native, but in computing "error" is often used as a verb AFAICT.

**Comment by svilupp** [comment]
Noted. In which case, I'll remove "error as a verb" fixes. It was distracting me during the read, but it seems to be only my problem.

**Comment by bkamins** [comment]
I would rely on the judgement of @pdeffebach who is a native speaker here. Though, maybe your proposals are better for non-natives? @pdeffebach - what do you think?

File: docs/src/index.md (hunk 1 above)
Comment (svilupp) [reply docs/src/index.md]: Updated as per @pdeffebach suggestion.

File: docs/src/index.md (hunk 2 above)
Comment (svilupp) [reply docs/src/index.md]: Updated as per @pdeffebach suggestion.

File: docs/src/index.md (hunk 3 above)
Comment (svilupp) [reply docs/src/index.md]: Updated as per @pdeffebach suggestion.

**Comment by svilupp** [comment]
Already followed @pdeffebach 's suggestion - left only the fix from "to" --> "do", which is likely to have been a typo.

Apologies for wasting your time (and huge thank you for the amazing packages!!)

**Comment by pdeffebach** [comment]
Thanks! Yes `error` as a verb is common, though I'm sure Milan actually wrote this.

**Pull request merged**